### PR TITLE
Add tooling to keep track of code coverage 

### PR DESCRIPTION
**Pull request by trevershick**
## Add jacoco coverage plugin to POM

Add the jacoco plugin so that it runs during 'mvn test'.  It will also generate a coverage report when 'mvn site' is run provided 'mvn test' has been run and target/jacoco.exec exists.

Addresses Bekreth/jetedge#86

## Fix FieldPojoAnalyzerTest when run with jacoco

When Jacoco runs and gathers coverage it creates a $jacocoData on classes it's testing.  This failed the FieldPojoAnalyzerTest because it didn't know to ignore this field.  Now it does.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -107,6 +108,40 @@
 
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.15</version>
```
Comment (Bekreth) [inline pom.xml]: Move the plugin version up to properties with all other versioning information.

**Comment by trevershick** [comment]
Yep. I’ll take care of it.  Just missed it I got all the others. :)

File: pom.xml (hunk 1 above)
Comment (trevershick) [reply pom.xml]: Addressed in a8984df

**Comment by trevershick** [comment]
Finally fixed this.  $jacocoData broke one of my tests.

**Pull request merged**